plugins/org.jkiss.dbeaver.data.transfer/src/org/jkiss/dbeaver/tools/transfer/stream/exporter/DataExporterJSON.java

```diff
@@ -160,19 +139,27 @@ public void exportRow(DBCSession session, DBCResultSet resultSet, Object[] row)
     private boolean isJsonDocumentResults(@NotNull Object[] row) {
         if (!ArrayUtils.isEmpty(columns)) {
             DBPDataKind dataKind = columns[0].getDataKind();
-            if (columns.length == 1 && (dataKind == DBPDataKind.DOCUMENT || dataKind == DBPDataKind.STRUCT)) { // STRUCT Kind - this is the case from Couchbase, it can contains JSON document also
-                if (row.length > 0 && !DBUtils.isNullValue(row[0]) && row[0] instanceof DBDDocument) {
-                    DBDDocument document = (DBDDocument) row[0];
+            if (columns.length == 1 &&
+                // STRUCT Kind - this is the case from Couchbase, it can contains JSON document also
```

Comment:
```suggestion
                // STRUCT Kind - this is the case from Couchbase, it can contain JSON documents as well
```

Reply:
TBH this is a bold statement. It can affect various databases in unexpected places